lesson05/main.js

```diff
@@ -1,17 +1,22 @@
-const ul = document.getElementById("js-ul"); 
-const fragment = document.createDocumentFragment(); 
 const elements = [ 
   {to: "bookmark.html", img:"img/1.png", alt:"画像1", text: "ブックマーク"}, 
   {to: "message.html", img:"img/2.png", alt:"画像2", text: "メッセージ"} 
 ]; 
-elements.forEach(function(element){ 
-  const li = document.createElement("li"); 
-  const a = document.createElement("a"); 
-  const img = document.createElement("img"); 
-  a.href = element.to; 
-  a.textContent = element.text; 
-  img.src = element.img; 
-  img.alt = element.alt; 
-  fragment.appendChild(li).appendChild(a).insertAdjacentElement("afterbegin",img); 
-}); 
-ul.appendChild(fragment);
+
+new Promise((resolve)=> {
+  resolve(elements);
+}).then((elements)=> {

```

Comment:
nits
For `elements` defined in a global variable, it is possible to make it a local variable.
By reducing the scope of variables, the code becomes more maintainable.
```suggestion
new Promise((resolve)=> {
  const elements = [ 
    {to: "bookmark.html", img:"img/1.png", alt:"画像1", text: "ブックマーク"}, 
    {to: "message.html", img:"img/2.png", alt:"画像2", text: "メッセージ"} 
  ]; 
  resolve(elements);
}).then((elements)=> {

```

Reply:
Thank you mai-san :)
From now on, I'll care about scope.